app/boards/arm/nrf52840_m2/nrf52840_m2.dts

```diff
@@ -0,0 +1,97 @@
+/*
+ * Copyright (c) 2020 The ZMK Contributors
+ * SPDX-License-Identifier: MIT
+ */
+
+/dts-v1/;
+#include <nordic/nrf52840_qiaa.dtsi>
+
+/ {
+	model = "Makerdiary nRF52840 M.2 module";
+	compatible = "makerdiary,nrf52840_m2";
+
+	chosen {
+		zephyr,code-partition = &code_partition;
+		//zephyr,console = &uart0;
+		//zephyr,bt-mon-uart = &uart0;
+		//zephyr,bt-c2h-uart = &uart0;
+		zephyr,sram = &sram0;
+		zephyr,flash = &flash0;
+	};
+
+	leds {
+		compatible = "gpio-leds";
+		red_led: led_0 {
+			gpios = <&gpio0 30 GPIO_ACTIVE_HIGH>;
+			label = "Red LED";
+		};
+		green_led: led_1 {
+			gpios = <&gpio0 29 GPIO_ACTIVE_HIGH>;
+			label = "Green LED";
+		};
+		blue_led: led_2 {
+			gpios = <&gpio0 31 GPIO_ACTIVE_HIGH>;
+			label = "Blue LED";
+		};
+	};
+
+};
+
+&gpio0 {
+	status = "okay";
+};
+
+&gpio1 {
+	status = "okay";
+};
+
+&uart0 {
+	compatible = "nordic,nrf-uart";
+	status = "okay";
+	current-speed = <115200>;
+	tx-pin = <16>;
+	rx-pin = <15>;
+	rts-pin = <14>;
+	cts-pin = <13>;
+};
+
+&usbd {
+	compatible = "nordic,nrf-usbd";
+	status = "okay";
+};
+
+
+&flash0 {
```

Comment:
I think I saw mention of additional flash on this? Not needed for the initial PR, but if it does have extra flash, we should get it defined at some point.